Collection assets to core

cholmes opened the pull request:
**Related Issue(s):** #1002 


**Proposed Changes:**

1. Updated Collection spec to include Assets
2. Removed Collection-level extension

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [ ] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [ ] This PR affects the [STAC API spec](https://github.com/radiantearth/stac-api-spec), and I have opened issue/PR #XXX to track the change.

cholmes commented:
Ok, I think this is ready for review. @m-mohr - I hacked through schema changes, so hopefully I did them right. Thought I got myself into a failing state, but I think I figured it out. But would be good for you to check it over.

I did click on 'this is not a breaking change', because I think it is in a very small way? If implementations said collection-assets for the extension then that will fail with it gone (though that will be true for every extension we move out).

m-mohr commented on CHANGELOG.md:
```diff
@@ -32,6 +33,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Removed
 
 - Checksum extension (field `checksum:multihash`). Use File Info extension (field `file:checksum`) instead for assets. There's no replacement for links.
+- Collection Assets extension, as the core construct of Assets in a Collection is now part of the core Collection spec.
```
Maybe add that no change is required except removing "collection-assets" from stac_extensions?

m-mohr reviewed (approved):
I've made a couple of changes:
- Remove extension from Code Owners file
- Align collection and item spec a bit more (i.e. moved assets details to "Additional Fields Information")
- As it was the last spec where it was missing, added h4 headers for "Additional Fields Information" in Items
- Add assets to examples (which may conflict with #955, but then it reminds us to check whether there are collection assets in the examples an add them if not)

Please also review my changes @cholmes and feel free to change things again.

cholmes commented:
Cool. Just looked it over, changes look good. A couple comments:

> As it was the last spec where it was missing, added h4 headers for "Additional Fields Information" in Items

I think I made these changes in another PR too, but we can just merge here and I'll deal.

> Add assets to examples (which may conflict with #955, but then it reminds us to check whether there are collection assets in the examples an add them if not)

#955 got merged - I added #1013 to track this though, as I agree we need to be sure to get them in.
